### DISCUSSION TITLE
Tests entropy consumption?

### DISCUSSION THREAD

**Discussion by beldmit** [open-quantum-safe/liboqs#1511] [General]
I see that the tests of 0.8.0 tend to significantly and sporadically hang on testing. I have an impression that it's related to the lack of entropy (running `ping 1.1.1.1` in a separate window makes the process much more stable and fast. 

I'm not familiar with the PQ crypto - am I correct? Can we do anything with this issue if I am correct?

**Comment by baentsch** [comment]
What platform are you testing on? Maybe the platform's demands on entropy are "special"? Something embedded (or otherwise underutilized? :-) Also, have you already tried a different randomness provider than the default (See `OQS_randombytes_custom_algorithm`)? Assuming you're using OpenSSL(?) does that perform rand generation OK standalone?

**Comment by beldmit** [comment]
Fedora 37, my workstation :)
I didn't try any custom randomness providers yet

**Comment by baentsch** [comment]
> Fedora 37

Hmm -- maybe that OS has/is configured for particular(ly stringent) entropy (quality) tests leading to randomness being generated more slowly than on other platforms? NIST SP800-22 gone overboard? Have you looked for/used (cross-platform) tests to check RNG performance to check  this is an O(Q:)S-specific issue?
